src/components/table/template.test.js

```diff
@@ -0,0 +1,332 @@
+
+      expect($componentHead.length).toEqual(1)
+      expect($componentHeadRow.length).toEqual(1)
+      expect($componentHeadCell.length).toEqual(noOfComponentHeadCells)
+    })
+
+    it('renders a caption', () => {
+      const args = examples['table-with-caption-and-head']
+      const { $ } = render('table', args)
+
+      const $component = $('.govuk-c-table')
+      const $componentCaption = $component.children('caption')
+      const captionContents = args.caption
+      const captionSize = args.captionSize
+
+      expect($componentCaption.length).toEqual(1)
+      expect($componentCaption.text()).toBe(captionContents)
+      expect($componentCaption.hasClass(captionSize)).toBeTruthy()
+    })
+  })
+
+  it('renders with classes', () => {
+    const args = {
+      'classes': 'custom-class-goes-here',
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    }
+    const { $ } = render('table', args)
+    const $component = $('.govuk-c-table')
+    expect($component.hasClass(args.classes)).toBeTruthy()
+  })
+
+  it('renders with attributes', () => {
+    const args = {
+      'attributes': {
+        'attribute-1': 'yes',
+        'attribute-2': 'no'
+      },
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric'
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    }
+    const { $ } = render('table', args)
+    const attributes = args.attributes
+    const $component = $('.govuk-c-table')
+
+    expect($component.attr('attribute-1')).toEqual(attributes['attribute-1'])
+    expect($component.attr('attribute-2')).toEqual(attributes['attribute-2'])
+  })
+
+  it('renders cell `colspan` attribute correctly', () => {
+    const args = {
+      'rows': [
+        [
+          {
+            'text': 'January'
+          },
+          {
+            'text': '£85',
+            'format': 'numeric',
+            'colspan': 2
+          }
+        ],
+        [
+          {
+            'text': 'February'
+          },
+          {
+            'text': '£165',
+            'format': 'numeric'
+          }
+        ]
+      ]
+    }
+    const { $ } = render('table', args)
+    const $component = $('.govuk-c-table')
+    const $componentBody = $component.children('tbody')
+    const $componentBodyRow = $componentBody.children('tr')
+
+    expect(parseInt($componentBodyRow.eq(0).children().eq(1).attr('colspan'), 10)).toBe(args.rows[0][1].colspan)
```

Comment:
If we used `toEqual` here, could we get rid of the casting?

Reply:
hmmm, with toEqual:

 Expected value to equal:
      2
 Received:
     "2"